### PR TITLE
Show number of images in progress bar

### PR DESCRIPTION
**Pull request by philippmwirth**
# Show number of images in progress bar

Show the number of images instead of the number of batches when embedding.

### CODE REVIEW

File: lightly/embedding/embedding.py [hunk 1]
```diff
@@ -107,20 +107,23 @@ def embed(self,
...
         efficiency = 0.0
         embeddings = []
         labels = []
         with torch.no_grad():
 
             start_timepoint = time.time()
-            for (img, label, fname) in pbar:
+            for (img, label, fname) in dataloader:
+
+                bsz = img.shape[0]
```
Comment (japrescott) [inline lightly/embedding/embedding.py:126]: why are you pulling out the with of an image? 🤔

Comment (guarin) [reply lightly/embedding/embedding.py:126]: `img` is a batch of images so `img.shape[0]` gives the batch size. But this line is actually not necessary as `batch_size = img.shape[0]` is already defined a bit further down (line 132)

File: lightly/embedding/embedding.py [hunk 2]
```diff
@@ -107,20 +107,23 @@ def embed(self,
...
+        )
 
         efficiency = 0.0
         embeddings = []
         labels = []
         with torch.no_grad():
 
             start_timepoint = time.time()
-            for (img, label, fname) in pbar:
+            for (img, label, fname) in dataloader:
```
Comment (MalteEbner) [inline lightly/embedding/embedding.py:124]: Rename this to images, labels, fnames.

File: lightly/embedding/embedding.py (hunk 1 above)
Comment (MalteEbner) [reply lightly/embedding/embedding.py:126]: Because img is actually a list of images (one batch), but badly named.

File: lightly/embedding/embedding.py [hunk 3]
```diff
@@ -107,20 +107,23 @@ def embed(self,
...
         if lightly._is_prefetch_generator_available():
-            pbar = tqdm(
-                BackgroundGenerator(dataloader, max_prefetch=3),
-                total=len(dataloader)
-            )
-        else:
-            pbar = tqdm(dataloader, total=len(dataloader))
+            dataloader = BackgroundGenerator(dataloader, max_prefetch=3),
+        
+        pbar = tqdm(
```
Comment (MalteEbner) [inline lightly/embedding/embedding.py:112]: You don't need to provide the dataloader as argument, see the example here:
https://github.com/tqdm/tqdm#manual

File: lightly/embedding/embedding.py (hunk 1 above)
Comment (MalteEbner) [reply lightly/embedding/embedding.py:126]: Furthermore, this line is duplicate to `batch_size = img.shape[0]` a few lines below -> delete it.

**Review by MalteEbner — COMMENTED** [review]
The logic is fine, but the naming can be improved.

File: lightly/embedding/embedding.py [hunk 4]
```diff
@@ -107,20 +107,23 @@ def embed(self,
...
+            dataloader,
```
Comment (MalteEbner) [inline lightly/embedding/embedding.py:113]: You don't need to provide the dataloader as argument, see the example here:
https://github.com/tqdm/tqdm#manual

**Review by MalteEbner — APPROVED** [review]
LGTM
